### DISCUSSION TITLE
How do I give data to not-found.tsx?

### DISCUSSION THREAD

**Discussion by emmmarosewalker** [vercel/next.js#46904] [Help]
### Summary

I need my not-found.tsx component to show different data in different scenarios, e.g. if the user is authed or not. Would I need to re-fetch the data I just fetched in page.tsx in not-found.tsx to accomplish this?

**Comment by icyJoseph** [comment]
I am not sure this supported just yet... or if it ever will, though it'd be nice to call `notFound(unknownData)` and be able to collect that from the `not-found` component, however, the docs state:

> not-found.js components do not accept any props.

If we judge the purpose of a not-found page, perhaps it is not the thing for your use case?

Suppose you have:

```jsx
// pseudocode - inside a server component

const auth = await getAuthFromCookies(cookies)

if (!auth) { notFound() }

return <UI />

```

Wouldn't it be more fitting to do:

```jsx
// pseudocode - inside a server component

const auth = await getAuthFromCookies(cookies)

if (!auth) return <MissingAuthUI />

return <UI />

```

After all the magic of Server Components is that, only the resulting HTML is sent to the client, roughly speaking, so you can import as much as you want within them, without worrying too much about client payload.

This way you also skip invoking the `not-found` page, which has some implications, like, it returns 404 to the client, and that's more like NotFound rather than 401 or 403 error codes.
